pkg/operator/operator.go

```diff
@@ -225,3 +236,8 @@ func kubeDNSIP(ctx context.Context, kubernetesInterface kubernetes.Interface) (n
 	}
 	return kubeDNSIP, nil
 }
+
+func setDurationAndExpiry(provider *stscreds.AssumeRoleProvider, ctx context.Context) {
+	provider.Duration = time.Duration(settings.FromContext(ctx).CredsDuration) * time.Minute
+	provider.ExpiryWindow = time.Duration(10) * time.Second
```

Comment:
Can you elaborate a little bit on this choice?

Reply:
From the documentation:

```
	// ExpiryWindow will allow the credentials to trigger refreshing prior to
	// the credentials actually expiring. This is beneficial so race conditions
	// with expiring credentials do not cause request to fail unexpectedly
	// due to ExpiredTokenException exceptions.
```

This is not a required change, I can delete it if you'd prefer.

Reply:
Hmm -- what's the default of the SDK. Is 10 seconds enough? Maybe we could choose something like 10% of the duration?